### PR TITLE
Create object stores dynamically

### PR DESCRIPTION
**Pull request by pashyaaa**
Added methods in the services which will help create object stores dynamically.

### CODE REVIEW

**Review by digvijay-gavas — APPROVED** [review]
This change is needed to dynamically create  object-store/s. The code looks good.

**Review by assuncaocharles — COMMENTED** [review]
The code looks good, can you add documentation on the readme?

**Comment by pashyaaa** [comment]
yeah sure..

File: projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts [hunk 1]
```diff
@@ -33,6 +33,41 @@ export class NgxIndexedDBService {
 		}
 	}
 
+	createObjectStores(
+		storeSchemas: ObjectStoreMeta[],
+		migrationFactory?: () => { [key: number]: (db: IDBDatabase, transaction: IDBTransaction) => void }
+	) {
+		if (migrationFactory != null) {
```
Comment (assuncaocharles) [inline projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts:40]: No need for the check here 

```
CreateObjectStore(this.indexedDB, this.dbConfig.name, this.dbConfig.version, storeSchemas);
```

Would be enough

File: projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts [hunk 2]
```diff
@@ -33,6 +33,41 @@ export class NgxIndexedDBService {
...
+				this.dbConfig.version,
+				storeSchemas,
+				migrationFactory
+			);
+		} else {
+			CreateObjectStore(this.indexedDB, this.dbConfig.name, this.dbConfig.version, storeSchemas);
+		}
+	}
+
+	createObjectStore(
```
Comment (assuncaocharles) [inline projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts:36]: Duplicated method, you can make it work with only one of them. Keep it DRY

File: projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts [hunk 3]
```diff
@@ -33,6 +33,41 @@ export class NgxIndexedDBService {
...
+			CreateObjectStore(
+				this.indexedDB,
+				this.dbConfig.name,
+				this.dbConfig.version,
+				storeSchemas,
+				migrationFactory
+			);
+		} else {
+			CreateObjectStore(this.indexedDB, this.dbConfig.name, this.dbConfig.version, storeSchemas);
+		}
```
Comment (assuncaocharles) [inline projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts]:
```suggestion
		CreateObjectStore(
				this.indexedDB,
				this.dbConfig.name,
				this.dbConfig.version,
				storeSchemas,
				migrationFactory
			);
```

File: projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts (hunk 2 above)
Comment (pashyaaa) [reply projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts:36]: Will delete this one. Looks redundant.

**Pull request merged**